evo1/atb/base.py

```diff
@@ -87,94 +21,98 @@ def _tap_confirm() -> None:
 
 # Handling of the actual battle logic itself (base class, replace with more complex logic)
 class SeqATBCombat(SeqBase):
-    def __init__(self, name: str = "Generic", wait_for_battle: bool = False) -> None:
+    # Finite state machine for keeping track of the battle state
+    class _BattleFSM(Enum):
+        PRE_BATTLE = auto()
+        BATTLE = auto()
+        POST_BATTLE = auto()
+
+    def __init__(self, name: str = "Generic") -> None:
         self.mem: BattleMemory = None
-        self.wait_for_battle = wait_for_battle
-        # Triggered is used together with wait_for_battle to handle boss sequences
-        self.triggered = False
+        self.state = self._BattleFSM.PRE_BATTLE
         super().__init__(name=name)
 
     def reset(self) -> None:
-        self.triggered = False
-
-    def execute(self, delta: float) -> bool:
-        if not self.update_mem():
-            if not self.wait_for_battle or self.triggered:
-                return True
-            # Tap confirm until battle starts
-            _tap_confirm()
-            return False
-        self.triggered = True
-
-        # Tap past win screen/cutscenes
-        if self.mem.ended:
-            _tap_confirm()
-        else:
-            self.handle_combat()
-
-        return not self.active
+        self.mem = None
+        self.state = self._BattleFSM.PRE_BATTLE
 
     def update_mem(self) -> bool:
         # Check if we need to create a new ATB battle structure, or update the old one
-        if self.mem is None:
-            self.mem = BattleMemory()
-        else:
-            self.mem.update()
+        self.mem = BattleMemory()
         # Clear unused memory; we need to try to recreate it next frame
         if not self.active:
             self.mem = None
             return False
         return True
 
-    # TODO: Predict hit-chance (need the formula based on acc/evade)
```

Comment:
This logic is now handled by the `AttackPrediction` class in `evo1/atb/predict.py`